vscode-extensions/vscode-spring-boot/lib/set-log-levels-ui.ts

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+import * as VSCode from 'vscode';
+import { LanguageClient } from "vscode-languageclient/node";
+import { ActivatorOptions } from '@pivotal-tools/commons-vscode';
+import { LiveProcess, LiveProcessLoggersUpdatedNotification, LiveProcessLogLevelUpdatedNotification, LiveProcessUpdatedLogLevel } from './notification';
+
+interface ProcessCommandInfo {
+    processKey : string;
+	label: string;
+	action: string;
+    projectName: string;
+}
+
+export interface Logger {
+    configuredLevel: string;
+    effectiveLevel: string;
+}
+
+export interface Loggers {
+    [propName: string]: Logger;
+}
+
+export interface LoggersData {
+    levels: string[];
+    loggers: Loggers;
+}
+
+export interface LoggerItem {
+    logger: Logger;
+    name: string;
+}
+
+async function setLogLevelHandler() {
+
+    const processData : ProcessCommandInfo[] = await VSCode.commands.executeCommand('sts/livedata/listProcesses');
+    const choiceMap = new Map<string, ProcessCommandInfo>();
+    const choices : string[] = [];
+    processData.forEach(p => {
+        const slash = p.action.lastIndexOf('/');
+        if (slash>=0) {
+            const choiceLabel = p.label;
+            choiceMap.set(choiceLabel, p);
+            choices.push(choiceLabel);
+        }
+    });
+    if (choices) {
+        const picked = await VSCode.window.showQuickPick(choices);
+        if (picked) {
+            const chosen = choiceMap.get(picked);
+            const loggers = await VSCode.commands.executeCommand('sts/livedata/getLoggers', chosen, {"endpoint": "loggers"});
+        }
+    }
+}
+
+async function getLoggersList(process: LiveProcess) {
+    const loggers: LoggersData = await VSCode.commands.executeCommand('sts/livedata/fetch/loggersData', process);
+    let items;
+    if(loggers) {
+        items = Object.keys(loggers.loggers).map(packageName => {
+            const logger: Logger = loggers.loggers[packageName];
+            const effectiveLevel = logger.effectiveLevel;
+            const label = packageName + ' (' + effectiveLevel + ')';
+            return {
+                packageName,
+                effectiveLevel,
+                label
+            };
+        });
+    }
+    if(items) {
+        const chosenPackage = await VSCode.window.showQuickPick(items);
+        if (chosenPackage) {
+            const chosenlogLevel = await VSCode.window.showQuickPick(loggers.levels);
+            const changeLogLevel = await VSCode.commands.executeCommand('sts/livedata/configure/logLevel', {"endpoint": "loggers"}, process, chosenPackage, {"configuredLevel":chosenlogLevel});
```

Comment:
Feels like `{"endpoint": "loggers"}` is a redundant piece of info.

Reply:
Updated getLoggers and removed caching logic. A progress bar is shown in the status bar of the window when loggers are being fetched.